### PR TITLE
Add check for nil otherTypeDef

### PR DESCRIPTION
**Pull request by farexw**
**Relation issue**
[e.g. https://github.com/swaggo/swag/pull/118/files](https://github.com/swaggo/swag/issues/1368)

This PR solve the problem

### CODE REVIEW

**Comment by farexw** [comment]
This PR solve the problem [issue](https://github.com/swaggo/swag/issues/1368)

**Comment by sdghchj** [comment]
Sorry,  I introduced this bug.

File: packages.go [hunk 1]
```diff
@@ -139,7 +139,7 @@ func (pkgDefs *PackagesDefinitions) parseTypesFromFile(astFile *ast.File, packag
 					fullName := typeSpecDef.TypeName()
 
 					anotherTypeDef, ok := pkgDefs.uniqueDefinitions[fullName]
-					if ok {
+					if ok && anotherTypeDef != nil {
```
Comment (sdghchj) [inline packages.go:142]: As follows:

```
if ok {
        if( anotherTypeDef == nil ){
                typeSpecDef.NotUnique = true
		pkgDefs.uniqueDefinitions[typeSpecDef.TypeName()] = typeSpecDef
        }else  if typeSpecDef.PkgPath != anotherTypeDef.PkgPath {
		anotherTypeDef.NotUnique = true
		typeSpecDef.NotUnique = true
		pkgDefs.uniqueDefinitions[fullName] = nil
		pkgDefs.uniqueDefinitions[anotherTypeDef.TypeName()] = anotherTypeDef
		pkgDefs.uniqueDefinitions[typeSpecDef.TypeName()] = typeSpecDef
	}
} else {
        pkgDefs.uniqueDefinitions[fullName] = typeSpecDef
}
```

Comment (sdghchj) [reply packages.go:142]: and also modify the function `parseFunctionScopedTypesFromFile`

**Review by sdghchj — CHANGES_REQUESTED** [review]

File: packages.go (hunk 1 above)
Comment (farexw) [reply packages.go:142]: Oк, I understand

Comment (farexw) [reply packages.go:142]: > and also modify the function `parseFunctionScopedTypesFromFile`

Oк, I understand

**Review by farexw — COMMENTED** [review]
ok

File: packages.go (hunk 1 above)
Comment (farexw) [reply packages.go]: Done, tank you for your attention

File: packages.go [hunk 2]
```diff
@@ -140,7 +140,10 @@ func (pkgDefs *PackagesDefinitions) parseTypesFromFile(astFile *ast.File, packag
 
 					anotherTypeDef, ok := pkgDefs.uniqueDefinitions[fullName]
 					if ok {
-						if typeSpecDef.PkgPath != anotherTypeDef.PkgPath {
+						if anotherTypeDef != nil {
```
Comment (sdghchj) [inline packages.go:143]: here the condition is ==

```
if anotherTypeDef == nil {

}
```

File: packages.go [hunk 3]
```diff
@@ -197,7 +200,10 @@ func (pkgDefs *PackagesDefinitions) parseFunctionScopedTypesFromFile(astFile *as
 
 								anotherTypeDef, ok := pkgDefs.uniqueDefinitions[fullName]
 								if ok {
-									if typeSpecDef.PkgPath != anotherTypeDef.PkgPath {
+									if anotherTypeDef != nil {
```
Comment (sdghchj) [inline packages.go:203]: the same as above

**Review by sdghchj — CHANGES_REQUESTED** [review]

File: packages.go (hunk 2 above)
Comment (farexw) [reply packages.go]: fixed

File: packages.go (hunk 3 above)
Comment (farexw) [reply packages.go]: fidxed

**Review by sdghchj — APPROVED** [review]
LGTM.
Thank you!

**Pull request merged**